src/bin/nydus-image/decompress.rs

```diff
@@ -0,0 +1,1158 @@
+extern crate tar;
+
+use std::{
+    collections::HashMap,
+    ffi::OsStr,
+    fs::{File, OpenOptions},
+    io::{self, Cursor, Error, ErrorKind, Read},
+    iter::{self, from_fn, repeat},
+    os::unix::prelude::{OsStrExt, OsStringExt},
+    path::{Path, PathBuf},
+    rc::Rc,
+    str,
+    sync::Arc,
+    vec::IntoIter,
+};
+
+use anyhow::Result;
+use nydus_api::http::LocalFsConfig;
+use nydus_rafs::{
+    metadata::{RafsInode, RafsMode, RafsSuper},
+    RafsIoReader,
+};
+use nydus_utils::compress::{self, Algorithm};
+use storage::{
+    backend::{localfs::LocalFs, BlobBackend, BlobReader},
+    device::{BlobChunkInfo, BlobInfo},
+    utils::alloc_buf,
+};
+use tar::{Builder, EntryType, Header};
+
+use crate::core::node::InodeWrapper;
+
+static PAX_SEP1: &[u8; 1] = b" ";
+static PAX_SEP2: &[u8; 1] = b"=";
+static PAX_PREFIX: &[u8; 13] = b"SCHILY.xattr.";
+static PAX_DELIMITER: &[u8; 1] = b"\n";
+
+pub trait Decompressor {
+    fn decompress(&self) -> Result<()>;
+}
+
+///  A decompressor with the ability to convert bootstrap file and blob file to tar
+pub struct OCIDecompressor {
+    bootstrap: Box<Path>,
```

Comment:
How about using `PathBuf` instead of `Box<Path>` here?

Reply:
> How about using `PathBuf` instead of `Box<Path>` here?

It's designed for unchangeable.  PathBuf is at lower cost, seems better.